### PR TITLE
Provide API for cloning of repositories.

### PR DESCRIPTION
**Pull request by smn**

### CODE REVIEW

**Comment by smn** [comment]
@Rizziepit @miltontony ready for review

File: unicore/distribute/api/validators.py [hunk 1]
```diff
@@ -32,3 +35,14 @@ def validate_schema(request):
     else:
         request.schema = schema.to_json()
         request.schema_data = data
+
+
+def repo_url_type_schema_validator(node, value):
+    valid_prefixes = ['git://', 'http://', 'https://', 'ssh://']
```
Comment (miltontony) [inline unicore/distribute/api/validators.py:41]: I recall having issues pushing using anything other than `https` (I think). Wouldn't this be a problem here too?

Comment (smn) [inline unicore/distribute/api/validators.py:41]: Possibly, but this is for cloning a repository. Regardless we need to sort that problem out and first verify it still is a problem. We're using a few releases later of GitPython now.

**Comment by miltontony** [comment]
:+1:

**Pull request merged**